src/main/java/org/jabref/logic/journals/JournalAbbreviationLoader.java

```diff
@@ -32,52 +35,75 @@
         return parser.getAbbreviations();
     }
 
-    public static List<Abbreviation> readJournalListFromFile(File file, Charset encoding) throws FileNotFoundException {
+    private static List<Abbreviation> readJournalListFromFile(Path file, Charset encoding) throws IOException {
         LOGGER.debug(String.format("Reading journal list from file %s", file));
         AbbreviationParser parser = new AbbreviationParser();
-        parser.readJournalListFromFile(Objects.requireNonNull(file), Objects.requireNonNull(encoding));
+        parser.readJournalListFromFile(file, Objects.requireNonNull(encoding));
         return parser.getAbbreviations();
     }
 
-    public void update(JournalAbbreviationPreferences journalAbbreviationPreferences) {
-        journalAbbrev = new JournalAbbreviationRepository();
-
-        // The order of reading the journal lists is important: last added abbreviation wins
-        // For instance, in the personal list one can overwrite abbreviations in the built in list
-
-        // Read builtin list
-        journalAbbrev.addEntries(readJournalListFromResource(JOURNALS_FILE_BUILTIN));
-
-        // Read IEEE list
-        if (journalAbbreviationPreferences.useIEEEAbbreviations()) {
-            journalAbbrev.addEntries(getOfficialIEEEAbbreviations());
-        } else {
-            journalAbbrev.addEntries(getStandardIEEEAbbreviations());
+    public static void writeDefaultDatabase(Path targetDirectory) {
+        try (MVStore store = MVStore.open(targetDirectory.resolve("journalList.mv").getParent().toString())) {
+            MVMap<String, String> fullToAbbreviation = store.openMap("FullToAbbreviation");
+
+            // Add all standard abbreviations
+            fullToAbbreviation.putAll(
+                    getBuiltInAbbreviations()
+                            .stream()
+                            .collect(Collectors.toMap(Abbreviation::getName, Abbreviation::getAbbreviation))
+            );
+
+            // Add all IEEE abbreviations
+            fullToAbbreviation.putAll(
+                    getStandardIEEEAbbreviations()
+                            .stream()
+                            .collect(Collectors.toMap(Abbreviation::getName, Abbreviation::getAbbreviation))
+            );
+
+            MVMap<String, String> abbreviationToFull = store.openMap("AbbreviationToFull");
+
+            // Add all standard abbreviations
+            abbreviationToFull.putAll(
+                    getBuiltInAbbreviations()
+                            .stream()
+                            .collect(Collectors.toMap(Abbreviation::getAbbreviation, Abbreviation::getName))
+            );
+
+            // Add all IEEE abbreviations
+            abbreviationToFull.putAll(
+                    getStandardIEEEAbbreviations()
+                            .stream()
+                            .collect(Collectors.toMap(Abbreviation::getAbbreviation, Abbreviation::getName))
+            );
+            //store.commit();
```

Comment:
Why is this disabled?

Reply:
It's not need apparently (happens automatically if the store is closed).